### DISCUSSION TITLE
Subscription already exists

### DISCUSSION THREAD

**Discussion by cholzer79** [abraunegg/onedrive#2500] [Q&A]
Today i noticed that onedrive did not sync anymore.
When I checked the logs I found this error:

```
Initializing the Synchronization Engine ...
Initializing monitor ...
OneDrive monitor interval (seconds): 300
Initializing subscription for updates ...
Webhook: handled validation request

ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 409 (Conflict)
  Error Reason:     Subscription Id c0bba80e-57a3-43a7-bac2-e6f525a76e7c already exists for the requested combination
  Error Timestamp:  2023-09-26T05:40:02
  API Request ID:   46ef4c56-5dca-40ce-bdc0-4d3dd9bbb939
ERROR: Unable to initialize subscriptions for updates. Please fix this issue.
```

Using the lastest https://hub.docker.com/r/driveone/onedrive

How do I fix this ? :)

**Comment by abraunegg** [comment]
@cholzer79 
> Using the lastest https://hub.docker.com/r/driveone/onedrive

Are you using 'latest' or 'edge' .... 'latest' is the latest main release, where as 'edge' is more current and follows GitHub commits.

> How do I fix this ? :)

Have you restarted your Docker container? Do you have two Docker containers with the *same* configuration trying to run at the same time?

When the service / Docker container is restarted, the existing subscription *should* be deleted. It should also then be *re-created* when it starts up.

**Reply by cholzer79** [reply]
> Are you using 'latest' or 'edge' .... 'latest' is the latest main release, where as 'edge' is more current and follows GitHub commits.

driveone/onedrive:latest

should I change this to "driveone/onedrive:edge" ? _(edit: same problem with edge)_

> Have you restarted your Docker container?

Yes, also restarted the machine, but all I get is:

```
Base Args: --monitor
# Launching onedrive
Configuration file successfully loaded
config file has been updated, checking if --resync needed
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Initializing monitor ...
OneDrive monitor interval (seconds): 300
Initializing subscription for updates ...
Webhook: handled validation request

ERROR: Microsoft OneDrive API returned an error with the following message:
  Error Message:    HTTP request returned status code 409 (Conflict)
  Error Reason:     Subscription Id c0bba80e-57a3-43a7-bac2-e6f525a76e7c already exists for the requested combination
  Error Timestamp:  2023-09-26T08:47:58
  API Request ID:   b087e1bb-f7ef-4dca-aa2e-019f5254e376
ERROR: Unable to initialize subscriptions for updates. Please fix this issue.
usermod: no changes
Base Args: --monitor
# Launching onedrive
Configuration file successfully loaded
Configuring Global Azure AD Endpoints
Initializing the Synchronization Engine ...
Initializing monitor ...
OneDrive monitor interval (seconds): 300
Initializing subscription for updates ...
Got termination signal, performing clean up
Shutting down the HTTP instance
Shutting down db connection and merging temporary data
```
I also already tried to delete the db and the token - reauthenticated but get the same "Subscription Id c0bba80e-57a3-43a7-bac2-e6f525a76e7c already exists for the requested combination" error.

> Do you have two Docker containers with the same configuration trying to run at the same time?

I have 2 containers (for well over 1 year), each using their own configuration, with separate static IPs and ports.
I stopped the 2nd one, to test if the first one would then work. That did not help.

In fact I get the same error for both instances now. O.o

**Comment by abraunegg** [comment]
@cholzer79 
Can you please open a bug report, and create a verbose debug log - very important as I will not be able to reproduce this.

I will need to create a PR, and you will then need to probably buuld manually Docker containers to test the PR, and provide feedback on the PR - and those steps might take a couple to fix this issue.

**Comment by abraunegg** [comment]
@cholzer79 
Please also note my timezone / location - if you can get the debug logs done overnight my time I can make inroads in looking at this in the morning.

**Comment by cholzer79** [comment]
> @cholzer79 Please also note my timezone / location - if you can get the debug logs done overnight my time I can make inroads in looking at this in the morning.

I created the report: https://github.com/abraunegg/onedrive/issues/2501

Since I am running this docker in Unraid, I would have to spin up a separate VM to be able to build a PR.
Any recommendation which Linux distro I should use for that? :)

**Reply by abraunegg** [reply]
Use any distro you are comfortable with
